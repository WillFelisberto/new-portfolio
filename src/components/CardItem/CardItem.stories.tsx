import { Meta, StoryObj } from '@storybook/react';
import CardItem from '.';

const meta: Meta<typeof CardItem> = {
  title: 'Components/CardItem',
  component: CardItem,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;

type Story = StoryObj<typeof CardItem>;

export const EducationExample: Story = {
  args: {
    logoUrl: '/ufsc.jpg',
    logoAlt: 'Education Logo',
    title: 'Computer Science',
    subtitle: 'Bachelor of Science',
    dateRange: 'Jan 2015 • Dec 2019 • (4 years)',
    description: {
      lexicalData: {
        root: {
          direction: 'ltr',
          format: '',
          indent: 0,
          type: 'root',
          version: 1,
          children: [
            {
              detail: 0,
              format: 0,
              mode: 'normal',
              style: '',
              text: 'Description',
              type: 'text',
              version: 1,
            },
          ],
        },
      },
    },
    linkUrl: 'https://university.edu',
    linkText: 'University Name',
  },
};

export const ExperienceExample: Story = {
  args: {
    logoUrl: '/ufsc.jpg',
    logoAlt: 'Company Logo',
    title: 'Company Name',
    subtitle: 'Software Engineer',
    dateRange: 'Feb 2020 • Present • (3 years)',
    description: {
      lexicalData: {
        root: {
          direction: 'ltr',
          format: '',
          indent: 0,
          type: 'root',
          version: 1,
          children: [
            {
              detail: 0,
              format: 0,
              mode: 'normal',
              style: '',
              text: 'Description',
              type: 'text',
              version: 1,
            },
          ],
        },
      },
    },
    linkUrl: 'https://company.com',
    linkText: '@ Company Name',
    additionalContent: (
      <div>
        <p className="mb-3 mt-6 text-sm font-semibold text-gray-400">Skills</p>
        <div className="mb-8 flex flex-wrap gap-x-2 gap-y-3 lg:max-w-[350px]">
          <span className="rounded bg-gray-600 px-2 py-1 text-white">
            React
          </span>
          <span className="rounded bg-gray-600 px-2 py-1 text-white">
            Node.js
          </span>
        </div>
      </div>
    ),
  },
};
