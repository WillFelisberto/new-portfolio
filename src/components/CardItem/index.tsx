import { fadeUpAnimation } from '@/lib/animations';
import { motion } from 'framer-motion';
import Image from 'next/image';
import RichTextComponent, { RichTextProps } from '../RichText';

export type CardItemProps = {
  logoUrl: string;
  logoAlt: string;
  title: string;
  subtitle?: string;
  dateRange: string;
  description?: Pick<RichTextProps, 'lexicalData'>; // RichText content
  linkUrl?: string;
  linkText?: string;
  additionalContent?: React.ReactNode;
};

const CardItem = ({
  logoUrl,
  logoAlt,
  title,
  subtitle,
  dateRange,
  description,
  linkUrl,
  linkText,
  additionalContent,
}: CardItemProps) => {
  return (
    <motion.div
      className="grid grid-cols-[40px,1fr] gap-4 md:gap-10"
      {...fadeUpAnimation}
      transition={{ duration: 0.5 }}
    >
      <div className="flex flex-col items-center gap-4">
        <div className="rounded-full border border-gray-500 p-0.5">
          <Image
            src={logoUrl}
            width={40}
            height={40}
            className="rounded-full"
            alt={logoAlt}
          />
        </div>
        <div className="h-full w-[1px] bg-gray-800" />
      </div>

      <div>
        <div className="flex flex-col gap-2 text-sm sm:text-base">
          {linkUrl && linkText && (
            <a
              href={linkUrl}
              target="_blank"
              aria-label={linkText}
              className="text-sky-100 transition-colors hover:text-sky-500"
              rel="noreferrer"
            >
              {linkText}
            </a>
          )}
          <h4 className="text-gray-300">{title}</h4>
          {subtitle && <p className="text-gray-400">{subtitle}</p>}
          <span className="text-gray-500">{dateRange}</span>
          {description && (
            <RichTextComponent lexicalData={description.lexicalData} />
          )}
        </div>

        {additionalContent && <div className="mt-4">{additionalContent}</div>}
      </div>
    </motion.div>
  );
};

export default CardItem;
