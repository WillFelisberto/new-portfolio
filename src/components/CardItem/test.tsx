import { render } from '@testing-library/react';

import CardItem from '.';

describe('<CardItem />', () => {
  const mockProps = {
    logoUrl: '/path/to/logo.png',
    logoAlt: 'Example Logo',
    title: 'Example Title',
    subtitle: 'Example Subtitle',
    dateRange: 'Jan 2020 • Dec 2023 • (3 years)',
    description: 'description',
    linkUrl: 'https://example.com',
    linkText: 'Example Link',
    additionalContent: <div>Additional Content</div>,
  };
  it('should render card with all required props correctly', () => {
    const { getByText, getByLabelText, getByAltText, container } = render(
      <CardItem {...mockProps} />,
    );

    expect(getByAltText(/Example Logo/i)).toBeInTheDocument();
    expect(getByText(/Example Title/i)).toBeInTheDocument();
    expect(getByLabelText(/Example Link/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
