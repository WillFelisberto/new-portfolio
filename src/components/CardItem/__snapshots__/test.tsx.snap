// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem /> should render card with all required props correctly 1`] = `
<div
  class="grid grid-cols-[40px,1fr] gap-4 md:gap-10"
  style="opacity: 0; transform: translateY(50px);"
>
  <div
    class="flex flex-col items-center gap-4"
  >
    <div
      class="rounded-full border border-gray-500 p-0.5"
    >
      <img
        alt="Example Logo"
        class="rounded-full"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fpath%2Fto%2Flogo.png&w=96&q=75"
        srcset="/_next/image?url=%2Fpath%2Fto%2Flogo.png&w=48&q=75 1x, /_next/image?url=%2Fpath%2Fto%2Flogo.png&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </div>
    <div
      class="h-full w-[1px] bg-gray-800"
    />
  </div>
  <div>
    <div
      class="flex flex-col gap-2 text-sm sm:text-base"
    >
      <a
        aria-label="Example Link"
        class="text-sky-100 transition-colors hover:text-sky-500"
        href="https://example.com"
        rel="noreferrer"
        target="_blank"
      >
        Example Link
      </a>
      <h4
        class="text-gray-300"
      >
        Example Title
      </h4>
      <p
        class="text-gray-400"
      >
        Example Subtitle
      </p>
      <span
        class="text-gray-500"
      >
        Jan 2020 • Dec 2023 • (3 years)
      </span>
      <div
        class="text-gray-400"
      />
    </div>
    <div
      class="mt-4"
    >
      <div>
        Additional Content
      </div>
    </div>
  </div>
</div>
`;
